hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/WriteJdbcP.java

```diff
@@ -0,0 +1,185 @@
+import java.util.List;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Private API, use {@link SinkProcessors#writeJdbcP}.
+ */
+public final class WriteJdbcP<T> implements Processor {
+
+    private static final IdleStrategy IDLER =
+            new BackoffIdleStrategy(0, 0, SECONDS.toNanos(1), SECONDS.toNanos(10));
+    private static final int BATCH_LIMIT = 50;
+
+    private final DistributedSupplier<Connection> connectionSupplier;
+    private final DistributedBiConsumer<PreparedStatement, T> bindFn;
+    private final String updateQuery;
+
+    private ILogger logger;
+    private Connection connection;
+    private PreparedStatement statement;
+    private List<T> itemList = new ArrayList<>();
+    private int idleCount;
+    private boolean supportsBatch;
+    private int batchCount;
+
+    private WriteJdbcP(
+            @Nonnull String updateQuery,
+            @Nonnull DistributedSupplier<Connection> connectionSupplier,
+            @Nonnull DistributedBiConsumer<PreparedStatement, T> bindFn
+    ) {
+        this.updateQuery = updateQuery;
+        this.connectionSupplier = connectionSupplier;
+        this.bindFn = bindFn;
+    }
+
+    /**
+     * Private API, use {@link SinkProcessors#writeJdbcP}.
+     */
+    public static <T> ProcessorMetaSupplier metaSupplier(
+            @Nonnull String updateQuery,
+            @Nonnull DistributedSupplier<Connection> connectionSupplier,
+            @Nonnull DistributedBiConsumer<PreparedStatement, T> bindFn
+
+    ) {
+        return ProcessorMetaSupplier.preferLocalParallelismOne(() ->
+                new WriteJdbcP<>(updateQuery, connectionSupplier, bindFn));
+    }
+
+    @Override
+    public void init(@Nonnull Outbox outbox, @Nonnull Context context) {
+        logger = context.logger();
+        connectAndPrepareStatement();
+    }
+
+    @Override
+    public void process(int ordinal, @Nonnull Inbox inbox) {
+        inbox.drainTo(itemList);
+        while (!itemList.isEmpty()) {
+            if (!reconnectIfNecessary()) {
+                continue;
+            }
+            try {
+                for (T item : itemList) {
+                    bindFn.accept(statement, item);
+                    addBatchOrExecute();
+                }
+                executeBatch();
+                connection.commit();
+                itemList.clear();
+                idleCount = 0;
+            } catch (Exception e) {
+                if (e instanceof SQLNonTransientException ||
+                        e.getCause() instanceof SQLNonTransientException) {
+                    throw ExceptionUtil.rethrow(e);
+                } else {
+                    logger.warning("Exception during update", e.getCause());
+                    idleCount++;
+                }
+            }
+        }
+    }
+
+    @Override
+    public boolean isCooperative() {
+        return false;
+    }
+
+    @Override
+    public void close() {
+        closeWithLogging(statement);
```

Comment:
looks like different closing approach with ReadJdbcP

Reply:
Yes, we are already logging while reconnecting instead of throwing the exception